### PR TITLE
fix(kit): add legacy entrypoints for pre v3.3 usage

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

https://github.com/nuxt/nuxt/issues/19717

### ❓ Type of change
- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

In various cases users on projects pre-3.3 who are consuming a 3.3 kit may receive an error message (because previously we were using cjs resolve utilities to resolve kit.

For example: https://github.com/nuxt/nuxt.new/pull/55.

This adds backward compatibility for those previous projects.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/kit/package.json [hunk 1]
```diff
@@ -4,6 +4,8 @@
   "repository": "nuxt/nuxt",
   "license": "MIT",
   "type": "module",
+  "main": "./dist/index.mjs",
+  "module": "./dist/index.mjs",
```
Comment (pi0) [inline packages/kit/package.json:8]: Module field is only picked by bundlers are you sure we need it?

Comment (danielroe) [reply packages/kit/package.json:8]: We might get away with just main 👍

Comment (danielroe) [reply packages/kit/package.json:8]: Oh dear. It's worse than that. We actually need a `require` export condition.

File: packages/kit/package.json [hunk 2]
```diff
@@ -8,7 +8,8 @@
   "exports": {
     ".": {
       "types": "./dist/index.d.ts",
-      "import": "./dist/index.mjs"
+      "import": "./dist/index.mjs",
+      "require": "./dist/index.mjs"
```
Comment (pi0) [inline packages/kit/package.json:12]: This is unsafe especially that i constantly see modules wrongly build with `.cjs` in dist and actually do `require('@nuxt/kit'`) but seems we don't have any choice at least until patched versions of nuxt 3 are widely adapted...

**Review by pi0 — APPROVED** [review]

File: packages/kit/package.json (hunk 2 above)
Comment (danielroe) [reply packages/kit/package.json:12]: oh, this is terrible, I agree. But I think `require` shouldn't work in any case, as we have `type: "module"`

**Pull request merged**